application/backend/src/main/resources/apps/typerefinery/components/graphs/barchart/clientlibs/functions.js

```diff
@@ -0,0 +1,204 @@
+                },
+                radius: 0,
+              },
+              plugins: [plugin],
+            });
+          
+            
+            graphItemsNs[componentConfig.resourcePath] = chartInstance;
+        }
+
+        ns.jsonDataConnected = async (dataSourceURL, $component) => {
+            try {
+                const response = await fetch(dataSourceURL).then((res) => res.json());
+                if (response) {
+                    ns.updateComponentHTML(response, $component);
+                    return;
+                }
+                ns.modelDataConnected($component);
+            }
+            catch (error) {
+                ns.modelDataConnected($component);
+            }
+        }
+
+        ns.tmsConnected = async (host, topic, $component) => {
+            try {
+                host = !host && "ws://localhost:8112";
+                if (!topic) {
+                    ns.modelDataConnected($component);
+                    return;
+                }
+                const componentData = localStorage.getItem(`${topic}`);
+                if (!componentData) {
+                    ns.modelDataConnected($component);
+                    return;
+                }
+                ns.updateComponentHTML(JSON.parse(componentData), $component);
+            }
+            catch (error) {
+                ns.modelDataConnected($component);
+            }
+        }
+
+        ns.modelDataConnected = ($component) => {
+            // Passing {} because, The values from the model obj are fetched in bellow function definition.
+            ns.updateComponentHTML({}, $component);
+        }
+
+        ns.dataReceived = (data, $component) => {
+            // Passing {} because, The values from the model obj are fetched in bellow function definition.
+            ns.updateComponentHTML(data, $component);
+        }
+
+        ns.init = ($component) => {
+            // parse json value from data-model attribute as component config
+            const componentConfig = componentNs.getComponentConfig($component);
+            const componentTopic = componentConfig.websocketTopic;
+            const componentHost = componentConfig.websocketHost;
+            const componentDataSource = componentConfig.dataSource;
+            console.log("componentDataSource",componentDataSource)
+            const componentPath = componentConfig.resourcePath;
+            console.log("componentConfig-1",componentConfig)
+            console.log("[barchart.js - functions.js] - Bar Chart Component");
+            // TMS.
+            if (componentHost && componentTopic) {
+                $component.setAttribute("id", componentTopic);
+                ns.tmsConnected(componentHost, componentTopic, $component);
+            }
+            // JSON
+            else if (componentDataSource) {
+                $component.setAttribute("id", componentPath);
+                ns.jsonConnected(componentDataSource, $component);
```

Comment:
There is nothing called jsonConnected.

Please do test before PR.

Reply:
Done